packages/inngest/src/components/InngestMetadata.test.ts

```diff
@@ -0,0 +1,228 @@
+import { afterEach, describe, expect, test, vi } from "vitest";
+import * as experimental from "../experimental";
+import type { Inngest } from "./Inngest.ts";
+import { buildTarget, UnscopedMetadataBuilder } from "./InngestMetadata.ts";
+
+const mockClient = () =>
+  ({
+    _updateMetadata: vi.fn().mockResolvedValue(undefined),
+  }) as unknown as Inngest;
+
+afterEach(() => {
+  vi.restoreAllMocks();
+});
+
+describe("buildTarget", () => {
```

Comment:
Fantastic tests! 🙌🙌🙌

Can we also add some type tests? e.g.

https://github.com/inngest/inngest-js/blob/94d80ab4ffbd4c72d35b0033e2afcee63b425f2b/packages/inngest/src/components/EventSchemas.test.ts#L33-L43

Vitest should already be set up to do this. Ideally we can simply test that the builder scoping works nicely, as that's the main complexity to make sure we don't break.

Some docs for it here: https://vitest.dev/guide/testing-types